src/ffi/dial_ffi.rs

```diff
@@ -248,14 +253,26 @@ pub extern "C" fn free_rust_runtime(rt_ptr: Option<Box<DialFfi>>) -> i32 {
             return -1;
         }
     };
-    match ctx.sigs.take() {
-        Some(sigs) => {
-            for sig in sigs {
-                let _ = sig.send(());
-            }
+    for channel in &ctx.channels {
+        let channel = match channel {
+            Either::A(chan) => chan.get_ref(),
+            Either::B(chan) => chan,
+        };
+        match channel {
+            ViamChannel::Direct(_) => (),
+            ViamChannel::WebRTC(chan) => ctx
+                .runtime
+                .as_ref()
+                .map(|rt| rt.block_on(async move { chan.close().await }))
+                .unwrap_or_default(),
         }
-        None => {}
     }
+    if let Some(sigs) = ctx.sigs.take() {
+        for sig in sigs {
+            let _ = sig.send(());
```

Comment:
Do we need the `let _ =` in the first place? What would happen if this line was just `sig.send(());`?

Reply:
clippy complains if we don't use the return value at all. doesn't need to be there (everything still compiles ofc), but I personally like this look better. probably best would be `let _: Result<(), ()> = sig.send(());` so we're forced to consider any type changes.